### ISSUE TITLE
Support for Angular 5

### ISSUE THREAD

**Issue by danielqa** [zoomsphere/ngx-store#34]
Upgrade to the newest version of Angular! +1

**Comment by HarelM** [comment]
I'm using this library with Angular 5 and except for the bug I'm having this works great.

**Comment by DanielKucal** [comment]
We also use it with Angular 5 without issues. @danielqa, what kind of problem are you experiencing?

**Comment by adammendoza** [comment]
I'm getting this error with Angular 5 
```
ERROR in node_modules/ngx-store/src/utility/storage/storage-event.d.ts(1,22): error TS2420: Class 'NgxStorageEvent' incorrectly implements interface 'StorageEvent'.
  Property 'NONE' is missing in type 'NgxStorageEvent'.
```

**Comment by adammendoza** [comment]
Based on this similar issue https://github.com/angular/angular-cli/issues/7474

I edited the utility.storage.storage-event.d.ts

But it also has an issue with the property oldValue being optional

```
ERROR in node_modules/ngx-store/src/utility/storage/storage-event.d.ts(1,22): error TS2420: Class 'NgxStorageEvent' incorrectly implements interface 'StorageEvent'.
  Property 'oldValue' is optional in type 'NgxStorageEvent' but required in type 'StorageEvent'.
```

Proposed changes: in lines 6 and 9
```
export declare class NgxStorageEvent implements StorageEvent {
    type: string;
    key: string;
    storageArea: Storage;
    protected static initTimeStamp: number;
    oldValue: any;
    newValue: any;
    timeStamp: number;
    NONE: any;
    readonly bubbles: boolean;
    readonly cancelBubble: boolean;
    readonly cancelable: boolean;
    readonly composed: boolean;
    readonly currentTarget: Window;
    readonly defaultPrevented: boolean;
    readonly eventPhase: number;
    readonly isTrusted: boolean;
    readonly path: Window[];
    readonly returnValue: boolean;
    readonly srcElement: any;
    readonly target: Window;
    readonly url: string;
    isInternal: boolean;
    constructor(type: string, key: string, storageArea: Storage);
    /**
     * Methods below exist only to satisfy TypeScript compiler
     */
    readonly scoped: boolean;
    readonly initEvent: any;
    readonly preventDefault: any;
    readonly stopImmediatePropagation: any;
    readonly stopPropagation: any;
    readonly deepPath: any;
    readonly AT_TARGET: number;
    readonly BUBBLING_PHASE: number;
    readonly CAPTURING_PHASE: number;
}

```

**Comment by adammendoza** [comment]
I submitted a PR

**Comment by DanielKucal** [comment]
@adammendoza, you can just set `skipLibCheck` to `true` in your tsconfig.json for this kind of issues

**Comment by DanielKucal** [comment]
So if it was issue caused by newer TypeScript version, it's resolved since v1.4.2.
